### DISCUSSION TITLE
Docs: markdown example in docs should actually process markdown

### DISCUSSION THREAD

**Discussion by IcedQuinn** [xmake-io/xmake#1731] [General]
Currently the example copies a file and renames it to have a .html extension. While a useful example for showing off how to copy and rename outputs it does not actually produce any kind of markdown.

The example should show off running a markdown processor and putting the output in the desired location. It should probably also link to the necessary methods for that rule to depend on the inputs, since following the example results in an xmake file which does not actually track the inputs vs output timestamps.

I might be able to do this as a PR if I figure out the code that belongs there.

**Comment by waruqi** [comment]
.md is just an extension that I write casually to demonstrate the use of custom rules.

Of course, I also welcome you to help improve this document example.

**Comment by IcedQuinn** [comment]
There's a PR over on the -docs repo which changes it to process markdown + outputting a progress log + dependency tracking.
